packages/start-slicemachine/src/legacyMigrations/migrateSMJSON.ts

```diff
@@ -0,0 +1,165 @@
+import * as path from "node:path";
+import * as fs from "node:fs/promises";
+
+import chalk from "chalk";
+import semver from "semver";
+
+import { SliceMachineManager, SliceMachineConfig } from "@slicemachine/manager";
+
+import { locateFileUpward } from "../lib/locateFileUpward";
+import {
+	Framework,
+	FRAMEWORKS,
+	UNIVERSAL,
+	detectFramework,
+} from "./detectFramework";
+
+export const migrateSMJSON = async (
+	manager: SliceMachineManager,
+): Promise<void> => {
+	let smJSONPath: string | undefined;
+
+	try {
+		smJSONPath = await locateFileUpward("sm.json", { startDir: manager.cwd });
+	} catch (error) {
+		// noop - sm.json does not exists
+	}
+
+	// Exit if sm.json is not found
+	if (!smJSONPath) {
+		return;
+	}
+
+	const smJSON = JSON.parse(await fs.readFile(smJSONPath, "utf-8"));
+
+	// Exit if sm.json was migrated
+	if ("//" in smJSON && !("libraries" in smJSON)) {
+		// eslint-disable-next-line no-console
+		return console.log(`
+${chalk.bgYellow(` ${chalk.black("WARN")} `)} ${chalk.magenta(
+			"sm.json",
+		)} is deprecated!
+
+  - If you were importing values from this file, please now import
+    them from ${chalk.magenta("slicemachine.config.json")}.
+    You can safely delete this file after to suppress this warning.
+
+  - If you weren't, you can safely delete this file to suppress
+    this warning.
+`);
+	}
+
+	// TODO: Decide on how to handle really old migrations (1+ year)
```

Comment:
If possible, a good start might be to track this information somewhere? Maybe we should check with Come, but it seems like this information would be valuable to take these decisions in the future. Either in segments or Sentry

Reply:
We're tracking the Slice Machine version users are using, so we know if many people are stuck in an old version already.

Does that sound good to you? Not sure I get it otherwise 🤔

Reply:
You're right 😅 sorry, completely forgot that we would have that data elsewhere!